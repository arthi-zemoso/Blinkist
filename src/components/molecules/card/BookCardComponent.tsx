import React from 'react';
import { makeStyles } from "@material-ui/core/styles";
import { Card, CardContent,Typography,Grid } from '@material-ui/core';
import { AccessTime, PersonOutline, MoreHoriz } from '@material-ui/icons';
import Book from '../../../assets/Books/3.svg'

type CardProp={
id:number,
img?:string,
title?:string,
alt?:string,
author?:string,
timeRead:string,
numberRead:string,
toAdd:boolean,
status?:"Currently Reading"|"Finished Reading"|"null",
finishset?:()=>void,
changeStatus?:()=>void,
}
const useStyles = makeStyles((theme) => ({
    root:{
      width: "18rem",
      borderRadius: "3%",
      borderBottomRightRadius: "2%",
      borderBottomLeftRadius: "2%",
      display: "grid",
      background:"rgb(250,250,250)",
    },
    image: {
      width: "inherit",
      borderRadius: "inherit",
      borderBottomRightRadius: "unset",
      borderBottomLeftRadius: "unset"
    },
    bookButton:{
      padding: theme.spacing(1),
      cursor: "pointer",
      backgroundColor:"#2ce080",
      color: "black"
    }
  
  }));
  
const BookCardComponent = (props:CardProp) => {
  
    const styles = useStyles();
    return (
      
        <Card className={styles.root} variant="outlined"
       
        >
          <img
            className={styles.image}
            src={props.img}
            title={props.title}
            alt = {props.alt}
          />
          <CardContent>
            <Typography gutterBottom variant="subtitle1" style={{fontWeight: 'bold'}}>
                {props.title}
            </Typography>
            <Typography variant="subtitle1">
              {props.author}
            </Typography>
            <br/>
            <Grid container justifyContent="space-between" >
              <Grid item ><AccessTime fontSize="small"  style={{marginTop: -2}} />{props.timeRead}</Grid>
              <Grid item><PersonOutline fontSize="small"  style={{marginTop: -2}} />{props.numberRead}</Grid>
            </Grid>

          </CardContent>
          <Grid container justifyContent="space-between">
          <Grid item >
             
          </Grid>
            <Grid item ><MoreHoriz fontSize="small" style={{margin: "1rem"}} /></Grid>
          </Grid>

        </Card>  
      
      );
    
  };
  BookCardComponent.defaultProps={
    img: Book,
title:"Doesn't hurt to ask",
alt:"bring human",
author:"Trey Gowdy",
timeRead:"13-min read",
numberRead:"1.9k reads",
toAdd:false,
status:"null",


}

export default BookCardComponent;

/* sx={[{width:284}
            ,{'&:hover': {
                backgroundColor:'#F1F6F4',
              },
              '&:hover button':{
                  backgroundColor:"blueviolet",
                  color:"white"
              }
               <LibraryBtn title="lib"/>
            }]}
            */
