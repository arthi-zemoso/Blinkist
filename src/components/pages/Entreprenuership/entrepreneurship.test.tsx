import { render, screen,fireEvent} from "@testing-library/react";
import LibraryBtn from "../../molecules/button/librarybtn";
import "@testing-library/jest-dom";
import Entrepreneurship from "./entrepreneurship";
import BookCardComponent from '../../molecules/card/BookCardComponent'
import renderer from 'react-test-renderer'
import {shallow} from "enzyme"
import { configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

configure({ adapter: new Adapter() })
const book={
    "id": 1,
    "name":"Being Boss",
    "author":"Kathleen Shannon and Emily...",
     "time":"13 minute read",
     "status":"F"
}
it("Is entrypage is visible",async()=>{
    render(<Entrepreneurship></Entrepreneurship>);
    const card=screen.getByText("Trending Blinks");
    expect(card).toBeInTheDocument();
});
it("is library button visible",async()=>{
    render(<LibraryBtn title="Add to Library"></LibraryBtn>);
    const button=screen.getByText("Add to Library");
    expect(button).toBeInTheDocument();
});
it("first snapshot",()=>{
    const component =renderer.create(<BookCardComponent id={book.id} title={book.name} author={book.author}></BookCardComponent>);
    let tree=component.toJSON();
    expect(tree).toMatchSnapshot();
});

it("button work",()=>{
    const callback=jest.fn();
    render(<LibraryBtn title="Add to Library" onClick={callback}></LibraryBtn>);
    fireEvent.click(screen.getByText(/Add to Library/i));
    expect(callback).toHaveBeenCalledTimes(1);
   
    });
    it("is grid visible",async()=>{
        render(<Entrepreneurship/>);
        const grid=screen.getByRole("textbox");
        expect(grid).toBeInTheDocument();
    });
    it("is account working",async()=>{
        render(<Entrepreneurship/>);
        const acc=await screen.findByTestId("4");
        fireEvent.click(acc);
        expect(acc).toBeInTheDocument();
    });






