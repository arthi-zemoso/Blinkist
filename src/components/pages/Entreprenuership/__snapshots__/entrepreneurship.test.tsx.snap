// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`first snapshot 1`] = `
<div
  className="MuiPaper-root MuiCard-root makeStyles-root-1 MuiPaper-outlined MuiPaper-rounded"
>
  <img
    alt="bring human"
    className="makeStyles-image-2"
    src="3.svg"
    title="Being Boss"
  />
  <div
    className="MuiCardContent-root"
  >
    <h6
      className="MuiTypography-root MuiTypography-subtitle1 MuiTypography-gutterBottom"
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      Being Boss
    </h6>
    <h6
      className="MuiTypography-root MuiTypography-subtitle1"
    >
      Kathleen Shannon and Emily...
    </h6>
    <br />
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-justify-content-xs-space-between"
    >
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
          focusable="false"
          style={
            Object {
              "marginTop": -2,
            }
          }
          viewBox="0 0 24 24"
        >
          <path
            d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
          />
          <path
            d="M12.5 7H11v6l5.25 3.15.75-1.23-4.5-2.67z"
          />
        </svg>
        13-min read
      </div>
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
          focusable="false"
          style={
            Object {
              "marginTop": -2,
            }
          }
          viewBox="0 0 24 24"
        >
          <path
            d="M12 5.9c1.16 0 2.1.94 2.1 2.1s-.94 2.1-2.1 2.1S9.9 9.16 9.9 8s.94-2.1 2.1-2.1m0 9c2.97 0 6.1 1.46 6.1 2.1v1.1H5.9V17c0-.64 3.13-2.1 6.1-2.1M12 4C9.79 4 8 5.79 8 8s1.79 4 4 4 4-1.79 4-4-1.79-4-4-4zm0 9c-2.67 0-8 1.34-8 4v3h16v-3c0-2.66-5.33-4-8-4z"
          />
        </svg>
        1.9k reads
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-justify-content-xs-space-between"
  >
    <div
      className="MuiGrid-root MuiGrid-item"
    />
    <div
      className="MuiGrid-root MuiGrid-item"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
        focusable="false"
        style={
          Object {
            "margin": "1rem",
          }
        }
        viewBox="0 0 24 24"
      >
        <path
          d="M6 10c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm12 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm-6 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
        />
      </svg>
    </div>
  </div>
</div>
`;
